### PR TITLE
Fix participant types get deleted issue

### PR DESCRIPTION
**Pull request by jingyi-gao-ttd**
- Remove on delete CASCADE from participantToTypes and approvers
- Change upsert to use a transaction to update participant  types

### CODE REVIEW

File: src/api/services/participantsService.ts
Comment (alex-yau-ttd) [inline src/api/services/participantsService.ts:108]: this method is updating (deleting and/or inserting) the participant (request) types and also updating the name, siteId and status - to me, the name of the method indicates the request types are being updated, but not the other things. Maybe either split into multiple methods or use a different method name (e.g. `updateParticipantAndTypes`, though typically `and` means it should be multiple methods)?

Related question - does it have to be one transaction for any particular reason?

Comment (jingyi-gao-ttd) [reply src/api/services/participantsService.ts]: I think both options make sense, I updated the name and also the logic to update request types into a separate function. 

> does it have to be one transaction for any particular reason?

I want to make sure that updates to the account's name, status, and type all happen simultaneously. If, for any reason, something goes wrong during this process, I want the system to automatically revert to the original state, allowing the approver to try again. If we were to update these elements separately, there's a risk that the participant could end up in an inconsistent state if one update succeeds while another fails. For example, the participant status is marked as approved, but the participant types don't get updated, and this participant will disappear from the awaiting approval list, which approver has no way to fix it in the portal.

**Review by alex-yau-ttd — APPROVED** [review]

File: src/api/services/participantsService.ts
Comment (alex-yau-ttd) [reply src/api/services/participantsService.ts]: makes sense - thanks!

**Pull request merged**
